integration/dkg/dkg_emulator_test.go

```diff
@@ -142,25 +145,29 @@ func (s *DKGSuite) runTest(goodNodes int, emulatorProblems bool) {
 	groupPubKey, err := crypto.DecodePublicKey(crypto.BLSBLS12381, groupPubKeyBytes)
 	assert.NoError(s.T(), err)
 
+	tag := "some tag"
+	hasher := crypto.NewBLSKMAC(tag)
 	// create and test a threshold signature with the keys computed by dkg
 	sigData := []byte("message to be signed")
-	signers := make([]*signature.ThresholdProvider, 0, len(nodes))
+	beaconKeys := make([]crypto.PrivateKey, 0, len(nodes))
 	signatures := []crypto.Signature{}
-	indices := []uint{}
+	indices := []int{}
 	for i, n := range nodes {
-		priv, err := n.dkgState.RetrieveMyBeaconPrivateKey(nextEpochSetup.Counter)
+		// TODO: to replace with safeBeaconKeys
+		beaconKey, err := n.dkgState.RetrieveMyBeaconPrivateKey(nextEpochSetup.Counter)
 		require.NoError(s.T(), err)
+		// epochLookup := epochs.NewEpochLookup(n.State)
+		// beaconKeyStore := hotsignature.NewEpochAwareRandomBeaconKeyStore(epochLookup, n.safeBeaconKeys)
```

Comment:
This does not work yet, will update in a separate PR.

Reply:
Do you know why it's not working?

Reply:
Because the database does have the key. I still need to spend more time debugging this. For now, I'll just keep as is, and do in a a separate PR